### PR TITLE
Fix corner case of empty items array

### PR DESCRIPTION
**Pull request by jbalboni**
### Reasons for making this change

When upgrading from an old version I found that our code was creating array schemas in the shape shown in the test case, which was resulting in the `getDefaultFormState` function creating form data that had an array of `undefined`s, instead of using `additionalItems` as the item schema. This meant that required fields in the array item schema weren't triggering validation errors.

### Checklist

* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated docs if needed
  - [x] I've run `npm run cs-format` on my branch to conform my code to [prettier](https://github.com/prettier/prettier) coding style

### CODE REVIEW

File: test/utils_test.js [hunk 1]
```diff
@@ -241,6 +241,26 @@ describe("utils", () => {
           foo: 42,
         });
       });
+
+      it("should fill array with additional items schema when items is empty", () => {
```
Comment (jbalboni) [inline test/utils_test.js]: This is not a great test description, but I'm not sure what a better one would be.

Comment (edi9999) [reply test/utils_test.js]: I think it is quite ok, it describes what it is testing.

**Comment by edi9999** [comment]
To me this looks good, I suggest to merge this soon.

**Comment by epicfaace** [comment]
@glasserc Can you merge this?

**Comment by glasserc** [comment]
Sorry for the delay! Thanks!

**Pull request merged**